components/Discussion/DiscussionPreferences.js

```diff
@@ -59,151 +64,301 @@ DiscussionPreferences.propTypes = {
+    try {
+      await setDiscussionPreferences(formState.anonymity, formState.credential)
+      onClose()
+    } catch (error) {}
   }
 
-  render() {
-    const { t, credentials, rules, onClose } = this.props
+  const existingCredential = credentials.find(
+    c => c.description === state.credential
+  )
+  const isListedCredential = existingCredential && existingCredential.isListed
 
-    const anonymity = (() => {
-      switch (rules.anonymity) {
-        case 'ALLOWED':
-          return {
-            disabled: false,
-            value: this.state.anonymity
-          }
-        case 'ENFORCED':
-          return {
-            disabled: true,
-            value: true
-          }
-        case 'FORBIDDEN':
-          return {
-            disabled: true,
-            value: false
-          }
-        default: {
-          console.warn(
-            `DiscussionPreferencesForm: unknown anonymity permission: ${rules.anonymity}`
-          )
-          return {
-            disabled: true,
-            value: false
-          }
-        }
-      }
-    })()
+  const credentialSuggestions = credentials.filter(
+    c => c.description !== state.credential
+  )
 
-    const existingCredential = credentials.find(
-      c => c.description === this.state.credential
+  const previewData = useMemo(() => {
+    const credentialInSuggestions = credentials.find(
+      value => value.description === state.credential
     )
-    const isListedCredential = existingCredential && existingCredential.isListed
 
-    const credentialSuggestions = credentials.filter(
-      c => c.description !== this.state.credential
-    )
+    return {
+      credential: credentialInSuggestions || { description: state.credential },
+      name: state.anonymity ? t('discussion/displayUser/anonymous') : me.name,
+      portrait: !state.anonymity && me.portrait
+    }
+  }, [state.anonymity, state.credential, credentialSuggestions, me])
+
+  const suggestedCredentialButtonStyling = useMemo(
+    () =>
+      css({
+        all: 'unset',
+        cursor: 'pointer',
+        padding: 10,
+        '@media(hover)': {
+          ':hover': {
+            backgroundColor: colorScheme.getCSSColor('hover')
+          }
+        },
+        '&:not(:last-child)': {
+          borderBottomWidth: 1,
+          borderBottomStyle: 'solid',
+          borderBottomColor: colorScheme.getCSSColor('divider')
+        }
+      }),
+    [colorScheme]
+  )
 
-    return (
-      <div>
-        <OverlayToolbar onClose={onClose} />
-        <OverlayBody>
+  return (
+    <form
+      onSubmit={event => {
+        event.preventDefault()
+        handleSubmit(state)
+      }}
+    >
+      <OverlayToolbar
+        title={t('components/DiscussionPreferences/modalTitle')}
+        onClose={onClose}
+      />
+      <OverlayBody noPadding>
+        <div
+          {...styles.previewWrapper}
+          {...colorScheme.set('borderBottomColor', 'divider')}
+        >
           <Interaction.P>
```

Comment:
I'd remove the text here and put it into CommentHeaderProfile als a Labe
<img width="321" alt="Screenshot 2021-11-26 at 15 07 43" src="https://user-images.githubusercontent.com/20746301/143593154-aafb270b-5b1f-4ea5-8371-64918769b1e8.png">
l